File: AutoMix Software/AutoMix Data Management/AudioDataExtractionProxy.cpp
// Copyright (C) 2016-2017 LesProjecteurs - All Rights Reserved
// Maxime STEVENOT, Guillaume HANNES, Jordan ERNULT, Louis CARLIER, Pierre GABON
// 
// This file is part of AutoMix Software.
// 
// Unauthorized copying of this file, via any medium is strictly prohibited.
// You should have received a copy of the License along with this program.

#include "stdafx.h"
#include "AudioDataExtractionProxy.h"
#include "AudioDataExtraction.h"

namespace AutoMixDataManagement {

	AudioDataExtractionProxy::AudioDataExtractionProxy()
	{
		_realSubject = gcnew AudioDataExtraction();
	}

	void AudioDataExtractionProxy::extractData(System::ComponentModel::BackgroundWorker^ bw, TrackCollection^ trackCollection)
	{
		_realSubject->extractData(bw, trackCollection);
	}

}

File: Automix Data Management/Extraction/AudioDataExtractionProxy.cs
// Copyright (C) 2016 - 2018 LesProjecteurs
// This file is part of Automix Software licensed under MIT License.

using System.ComponentModel;
using Automix_Data_Management.Model;

namespace Automix_Data_Management.Extraction
{
    public class AudioDataExtractionProxy : IAudioDataExtraction
    {
        private readonly IAudioDataExtraction _realSubject;

        public AudioDataExtractionProxy()
        {
            _realSubject = new AudioDataExtraction();
        }

        public void ExtractData(BackgroundWorker backgroundWorker, TrackCollection trackCollection)
        {
            _realSubject.ExtractData(backgroundWorker, trackCollection);
        }
    }
}
